cpp/controller/cyclescontroller.cpp

```diff
@@ -0,0 +1,70 @@
+/********************************************
+**
+** Copyright 2016 JustCode Justyna Kulinska
+**
+** This file is part of Resto.
+**
+** Resto is free software; you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation; either version 2 of the License, or
+** any later version.
+**
+** Resto is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Resto; if not, write to the Free Software
+** Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+**
+********************************************/
+
+#include "cyclescontroller.h"
+
+CyclesController::CyclesController(SettingsController &settingsController, QObject *parent)
+    : QObject(parent), m_settingsController(settingsController)
+{
+    connect(this, &CyclesController::currentIntervalChanged, this, [this] { emit isCycleFinishedChanged(isCycleFinished()); });
+}
+
+int CyclesController::maxCycleIntervals() const
+{
+    return sc_maxCycleIntervals;
+}
+
+int CyclesController::currentInterval() const
+{
+    return m_settingsController.cyclesMode() ? m_currentInterval
+                                             : 0;
+}
+
+bool CyclesController::isCycleFinished() const
+{
+    return m_settingsController.cyclesMode()
+            ? (m_currentInterval == m_settingsController.cycleIntervals())
+            : 0;
+}
+
+void CyclesController::setCurrentInterval(int currentInterval)
+{
+    if (m_currentInterval == currentInterval)
```

Comment:
It seems to me you're using `{}` in single line ifs

Reply:
I guess it might be mixed here and there but yes, the newest style is to use. Thanks!